### ISSUE TITLE
Problems to understand the analysis of the tutorial 

### ISSUE THREAD

**Issue by Gregjlt** [saezlab/decoupleR#110]
Hello, 
I'm having trouble understanding the interpretation of your [example](https://saezlab.github.io/decoupleR/articles/tf_bk.html) for the transcription factor activity inference in bulk RNA-seq, particularly on this graph : 
![image](https://github.com/saezlab/decoupleR/assets/117280379/0f7e600c-49ec-49cc-881d-62c01e770bad)

Looking at the code below, genes in red, are the ones that have the same sign for the MoR and the t value, and in blue the ones that have opposite signs. So, for the best predicted TFs, I would expect to have the majority of genes that are represented in red, as this means that the logFC observed corresponds to the type of regulation predicted by the MoR. However, we can see that for SP1, one of the best predicted TFs, almost half of the genes are in blue, so have a t value in opposite sign with the MoR. Am I understanding something wrong in this example ? 
```R
tf <- 'SP1'

df <- net %>%
  filter(source == tf) %>%
  arrange(target) %>%
  mutate(ID = target, color = "3") %>%
  column_to_rownames('target')

inter <- sort(intersect(rownames(deg),rownames(df)))
df <- df[inter, ]
df[,c('logfc', 't_value', 'p_value')] <- deg[inter, ]
df <- df %>%
  mutate(color = if_else(mor > 0 & t_value > 0, '1', color)) %>%
  mutate(color = if_else(mor > 0 & t_value < 0, '2', color)) %>%
  mutate(color = if_else(mor < 0 & t_value > 0, '2', color)) %>%
  mutate(color = if_else(mor < 0 & t_value < 0, '1', color))

ggplot(df, aes(x = logfc, y = -log10(p_value), color = color, size=abs(mor))) +
  geom_point() +
  scale_colour_manual(values = c("red","royalblue3","grey")) +
  geom_label_repel(aes(label = ID, size=1)) + 
  theme_minimal() +
  theme(legend.position = "none") +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  geom_hline(yintercept = 0, linetype = 'dotted') +
  ggtitle(tf)
```
Thanks for your answer !

**Comment by PauBadiaM** [comment]
Hi @Gregjlt,

To compute enrichment scores we use the univariate linear model method:
![](https://decoupler-py.readthedocs.io/en/1.4.0/_images/ulm.png)
In this model where each gene is an observation, the response variable (y) is the change in gene expression and the explanatory variable (x) is the weight for that TF-Gene interaction. In the case of SP1, the red dots that have positive t-values belong to the first quadrant (+ and +), and the blue dots that have negative values belong to the 3rd quadrant (- and -). If you fit a line (like the one shown in the attached image) you would get a positive slope. The moment you have the contrary, blue dots with positive t-values (- and +) and/or red dots with negative t-values (+ and -) the slope becomes negative.

Basically, you can get a positive activity by having most positive targets with positive t-values, or by having most negative targets with negative t-values, or a combination of both. The moment this becomes inconsistent, for example, a random mixture of positive and negative targets with either positive or negative t-values, there will be no trend and the slope of the linear model will be flat.

Hope this is helpful! Let me know if you need any further clarification.

**Comment by Gregjlt** [comment]
Hi @PauBadiaM , 

Thanks a lot for the answer ! 
As I understand the Univaraite linear model you use, I still can't really get this plot and the explanation just below.
![image](https://github.com/saezlab/decoupleR/assets/117280379/2c264158-93f0-4f9f-a378-ff84905bccaf)

Indeed, I don't understand how blue can means that those genes are "deactivating" the TF. Looking at your regulation network, 
![image](https://github.com/saezlab/decoupleR/assets/117280379/f87c76db-d9d7-42a4-ba40-9a627ecccd38)

I thought that it took into account the influence of the TFs on the genes (either positive or negative, and represented by the variable MoR) but not the influence of the genes on the TFs. 
Therefore, to me, a blue gene like CLU for example means that this gene should have, according to the network, a "theoritical" positive logFC (or t-value as they have the same sign), but has in reality a negative logFC. That's why I don't understand the choice of colors for your graph, where blue dots with negative logFC seems to be well placed but are in reality not "correctly" predicted by the network. So here, I understand that red means that the genes are correctly placed (i.e a correct logfc sign) and the blue ones are wrongly placed (i.e a incorrect logfc, if we refer to the network).

**Comment by yaolin101** [comment]
Hi @PauBadiaM,

this boxplot is quite clear. But for the mouse, I see some regulons only has one value for the mor. For example, all the mor value for the interaction between Arntl and its target is 1. Then how does the ULM fit the line and get the slop?


Thanks,
Yao

**Comment by yaolin101** [comment]
And I have another question about how to understand the result.

 Let's take the Arntl transcription factor as an example.

**Arntl’s score from deconpleR::ulm:** Positive and significant.
**Interaction weights:** All  positive (1 for all the targets)
**Expression changes:** Arntl: Negative log2FC (TF itself is downregulated); Targets: Positive log2FC (targets are upregulated).



If I understand correctly, the positive score from the decoupleR::run_ulm comes from the positive interaction weights and the positive log2FC of its targets. 
But here the Arntl itself has negative log2FC. how is it possible and how do i interpret it? 

Thanks a lot. 

Yao
